x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -180,7 +183,31 @@ public Analyzer(AnalyzerContext context, Verifier verifier) {
 
     public LogicalPlan analyze(LogicalPlan plan) {
         BitSet partialMetrics = new BitSet(FeatureMetric.values().length);
-        return verify(execute(plan), gatherPreAnalysisMetrics(plan, partialMetrics));
+
+        List<LogicalPlan> analyzedPlans = new ArrayList<>();
+
+        LogicalPlan forkAnalyzed = plan.transformDown(Fork.class, fr -> {
+            int count = 1; // automatic fork branch ids start at 1
+            for (var subPlan : fr.subPlans()) {
+                LogicalPlan subPlanCopy = subPlan.transformUp(
+                    LogicalPlan.class,
+                    p -> p instanceof LeafPlan ? p : p.replaceChildren(p.children())
+                );
+
+                String forkValue = "fork" + count++;
+                subPlanCopy = new Eval(
+                    fr.source(),
+                    subPlanCopy,
+                    List.of(new Alias(fr.source(), "_fork", new Literal(fr.source(), forkValue, KEYWORD)))
+                );
+
+                LogicalPlan analyzedCopy = execute(subPlanCopy);
```

Comment:
Btw, you're skipping verification and metric gathering.
Double check by adding a few tests with failing queries.